### PR TITLE
Add flip function

### LINKED ISSUE #130: flip function
Similar to ramda's [flip](http://ramdajs.com/docs/#flip). Sometimes it is beneficial to flip the argument order of a function. Is this something that would be useful in this project?

I would be willing to implement this. Thoughts?

### PR DESCRIPTION
**Pull request by BrainMaestro**
Closes #130.

### CODE REVIEW

File: src/Functional/Flip.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * Return a version of the given function where the first arguments two arguments are swapped.
+ *
+ * Any other arguments that are supplied are provided in their correct order unmodified.
+ * If one argument is provided, it is passed to the function without change.
+ *
+ * @param callable $function the function you want to flip
+ * @return callable a flipped version of the given function
+ */
+function flip(callable $callback)
```
Comment (lstrojny) [inline src/Functional/Flip.php]: I didn't get that flip would only flip the first two previously and now I wonder about the name. Should it be `flip_2` or something instead?

Comment (BrainMaestro) [inline src/Functional/Flip.php]: Would you want it to flip all the arguments? I was just trying to stay as close to the original rambda implementation as possible

Comment (lstrojny) [inline src/Functional/Flip.php]: I don't have the need for `flip` or `flip_2` yet but I worry about future naming collisions if we use the generic name `flip` for something as specific as flipping the first two arguments. What do you think?

Comment (BrainMaestro) [inline src/Functional/Flip.php]: Okay. That makes sense. I could make this more generic by allowing the user to provide the number of arguments to flip. `flip($numArguments, $callback)`. That should handle the issue you mentioned.

Comment (BrainMaestro) [inline src/Functional/Flip.php]: I decided it would be more beneficial to just flip the argument order completely. I think that makes more sense.

**Comment by BrainMaestro** [comment]
Hey. Sorry to bother, but are you still interested in this?

**Comment by lstrojny** [comment]
@BrainMaestro sure, looks good now!

**Pull request merged**